src/daemon/instance_settings_handler.h

```diff
@@ -72,6 +72,12 @@ class InstanceSettingsException : public SettingsException
     InstanceSettingsException(const std::string& reason, const std::string& instance, const std::string& detail);
 };
 
+class InstanceStateSettingsException : public SettingsException
```

Comment:
Wouldn't it make more sense to derive from `InstanceSettingsException` (which has the exact same constructor code)?

Conceptually, an `InstanceStateSettingsException` is-a `InstanceSettingsException`. OTOH, when we deal with instance-related settings exceptions, we want to include `InstanceStateSettingsException` too (except when the handling for `InstanceStateSettingsException` is specific, but then we would place the correponding catch clause first).

Am I missing something? Is there any case where we want to handle `InstanceSettingsException` but not `InstanceStateSettingsException`?